### ISSUE TITLE
User Podman service is also available

### ISSUE THREAD

**Issue by bertmelis** [cockpit-project/cockpit-podman#703]
Host: Debian 10
Cockpit-podman 28-1 (build from source)

Every time I open the Podman containers page I have the option to start the "user podman service". However, this does not seem to be persistent. Every time I navigate away and return, I have to "re-enable" the user podman service:
![image](https://user-images.githubusercontent.com/23474336/114213030-c96ada80-9962-11eb-9045-6b03f85f6371.png)

I have no clue what it does.

**Comment by martinpitt** [comment]
What it does is that it checks/enables this:

    systemctl --user status podman.socket

Is this running for you? Is it enabled? Does it somehow re-disable itself (e.g. because of some error)?

When you press the "Start" button, does that actually succeed starting podman.socket? Can you see your user images/containers?

**Comment by bertmelis** [comment]
While the "start-button" is still present:

```
● podman.socket - Podman API Socket
   Loaded: loaded (/usr/lib/systemd/user/podman.socket; enabled; vendor preset: enabled)
   Active: active (listening) since Sat 2021-04-10 22:01:23 CEST; 13h ago
     Docs: man:podman-system-service(1)
   Listen: /run/user/1000/podman/podman.sock (Stream)
   CGroup: /user.slice/user-1000.slice/user@1000.service/podman.socket
```
I can see my containers although CPU and memory info is not available:
![image](https://user-images.githubusercontent.com/23474336/114298624-a56cdd80-9ab7-11eb-85f3-9cb1b41411cd.png)

**Comment by bertmelis** [comment]
aha, a clue here:
```
Request Failed(Internal Server Error): failed to obtain Container 2d47da687fea3e3b1949a8d7f692a77563f8eb21fbb2889e9de3f9764ba1e3b2 stats: unable to load cgroup at /user.slice/user-1000.slice/user@1000.service/user.slice/podman-32131.scope: no support for CGroups V1 in rootless environments
```

**Comment by marusak** [comment]
Can you please run these two command in terminal as normal user and post output here? The first one should hang (it is listening on upcoming events. Please leave it listening for ~2minutes and then you can ctrl+C it. If it does not hang that is the issue).

`curl --unix-socket /run/user/1000/podman/podman.sock http://d/v1.24/events`
`curl --unix-socket /run/user/1000/podman/podman.sock http://d/v1.24/libpod/info`

> aha, a clue here:

That is expected. That is why you have `n/a` in the UI. For cgroupsV1 these items are just not available.

**Comment by bertmelis** [comment]
`curl --unix-socket /run/user/1000/podman/podman.sock http://d/v1.24/events`

```
{"cause":"failed to get cursor: cannot assign requested address","message":"failed to get journal cursor: failed to get cursor: cannot assign requested address","response":500}
```
It doesn't hang indeed. I know it's not a cockpit-podman issue but any pointers are welcome. Feel free to close the issue here.

`curl --unix-socket /run/user/1000/podman/podman.sock http://d/v1.24/libpod/info`

```
{"host":{"arch":"amd64","buildahVersion":"1.19.4","cgroupManager":"cgroupfs","cgroupVersion":"v1","conmon":{"package":"conmon: /usr/libexec/podman/conmon","path":"/usr/libexec/podman/conmon","version":"conmon version 2.0.27, commit: "},"cpus":4,"distribution":{"distribution":"debian","version":"10"},"eventLogger":"journald","hostname":"server","idMappings":{"gidmap":[{"container_id":0,"host_id":1000,"size":1},{"container_id":1,"host_id":100000,"size":65536}],"uidmap":[{"container_id":0,"host_id":1000,"size":1},{"container_id":1,"host_id":100000,"size":65536}]},"kernel":"4.19.0-16-amd64","memFree":198455296,"memTotal":16453894144,"ociRuntime":{"name":"crun","package":"crun: /usr/bin/crun","path":"/usr/bin/crun","version":"crun version 0.18.1-7931a-dirty\ncommit: 7931a1eab0590eff4041c1f74e2844b297c31cea\nspec: 1.0.0\n+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL"},"os":"linux","security":{"apparmorEnabled":false,"capabilities":"CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT","rootless":true,"seccompEnabled":true,"selinuxEnabled":false},"slirp4netns":{"executable":"/usr/bin/slirp4netns","package":"slirp4netns: /usr/bin/slirp4netns","version":"slirp4netns version 1.1.8\ncommit: unknown\nlibslirp: 4.3.1-git\nSLIRP_CONFIG_VERSION_MAX: 3\nlibseccomp: 2.4.4"},"swapFree":1001648128,"swapTotal":1024454656,"uptime":"21h 40m 50.91s (Approximately 0.88 days)","linkmode":"dynamic"},"store":{"configFile":"/home/bert/.config/containers/storage.conf","containerStore":{"number":4,"paused":0,"running":4,"stopped":0},"graphDriverName":"overlay","graphOptions":{"overlay.mount_program":{"Executable":"/usr/bin/fuse-overlayfs","Package":"fuse-overlayfs: /usr/bin/fuse-overlayfs","Version":"fusermount3 version: 3.4.1\nfuse-overlayfs: version 1.4\nFUSE library version 3.4.1\nusing FUSE kernel interface version 7.27"}},"graphRoot":"/home/bert/.local/share/containers/storage","graphStatus":{"Backing Filesystem":"extfs","Native Overlay Diff":"false","Supports d_type":"true","Using metacopy":"false"},"imageStore":{"number":34},"runRoot":"/run/user/1000/containers","volumePath":"/home/bert/.local/share/containers/storage/volumes"},"registries":{"search":["docker.io","quay.io"]},"version":{"APIVersion":"3.0.0","Version":"3.0.1","GoVersion":"go1.14","GitCommit":"","BuiltTime":"Thu Jan  1 01:00:00 1970","Built":0,"OsArch":"linux/amd64"}}
```

**Comment by martinpitt** [comment]
BTW, Debian 10 is not officially supported by cockpit-podman, as podman itself is not in Debian 10. I suppose you got that from some third-party repo, and that may not package the user service correctly?

**Comment by bertmelis** [comment]
That's true, I build from source.

I've enabled cgroups v2 now. Now `curl --unix-socket /run/user/1000/podman/podman.sock http://d/v1.24/libpod/info` hangs

**Comment by bertmelis** [comment]
https://github.com/containers/podman/issues/8125

I'll check this tomorrow.

**Comment by marusak** [comment]
> Now `curl --unix-socket /run/user/1000/podman/podman.sock http://d/v1.24/libpod/info` hangs. But the cpu and memory are shown.

So it hangs? Is the alert still being shown?

**Comment by bertmelis** [comment]
Nono, I'm still at square one. Only difference is that I'm using cgroups v2 now.

**Comment by marusak** [comment]
Right. I am not able to help you much more then that. If your podman is not working, you might ask in https://github.com/containers/podman . But there might be a reason why it is not in debian 10.
Also we don't test this version so it may happen that c-podman won't fully work anyway. We test on debian 11 so that should be working.

**Comment by bertmelis** [comment]
I'll check over there. Thank you for the help.
Debian Bullseye is on it's way so I might not spend too much time on this as my full rootless containers are working.

**Comment by marusak** [comment]
> Is there a way to make the "start" permanent so clicking it all the time is not needed?

```
$ curl --unix-socket /run/user/1000/podman/podman.sock http://d/v1.24/events
curl: (18) transfer closed with outstanding read data remaining
$ 
```
^ This is problem with podman and unfortunately there is no workaround that can be done on our side to work around it.
I think that the only "solution" would be to update to newer podman (which may unfortunately not be available for 8.3)

> but without CPU/Memory information.

That is unrelated. That is due to using cgroupsV1.

**Comment by linickx** [comment]
Thanks very much 👍 

So both issues are the curse of enterprise linux.

To save someone in future a google, these 2x links seem to explain why cgroupsV1 is the default on RHEL8.
* https://www.redhat.com/en/blog/world-domination-cgroups-rhel-8-welcome-cgroups-v2
* https://access.redhat.com/articles/3735611
